plugins/modules/cloud/misc/proxmox_snap.py

```diff
@@ -137,14 +147,31 @@ def snapshot_remove(self, vm, vmid, timeout, snapname, force):
             time.sleep(1)
         return False
 
+    def snapshot_rollback(self, vm, vmid, timeout, snapname):
+        if self.module.check_mode:
+            return True
+
+        taskid = self.snapshot(vm, vmid)(snapname).post("rollback")
+        while timeout:
+            if (self.proxmox_api.nodes(vm['node']).tasks(taskid).status.get()['status'] == 'stopped' and
+                    self.proxmox_api.nodes(vm['node']).tasks(taskid).status.get()['exitstatus'] == 'OK'):
```

Comment:
How about rewriting this as:
```suggestion
            status = self.proxmox_api.nodes(vm['node']).tasks(taskid).status.get()
            if status['status'] == 'stopped' and status['exitstatus'] == 'OK':
```
Then it only needs to do one API request instead of two (assuming it doesn't do some very clever and dangerous caching).